src/main/java/org/springframework/cloud/deployer/spi/kubernetes/DefaultContainerFactory.java

```diff
@@ -171,6 +170,31 @@ public Container create(String appId, AppDeploymentRequest request, Integer port
 			}
 		}
 
+        if(properties.getReadinessProbePort() != null) {
+            container.withReadinessProbe(
+                    new ProbeCreator(properties.getReadinessProbePort(), properties.getReadinessProbePath(), properties.getReadinessProbeTimeout(),
+                            properties.getReadinessProbeDelay(), properties.getReadinessProbePeriod(), "readiness",
+                            request.getDeploymentProperties()).create());
+        }
+
+        if (properties.getLivenessProbePort() != null) {
+            container.withLivenessProbe(
+                    new ProbeCreator(properties.getLivenessProbePort(), properties.getLivenessProbePath(), properties.getLivenessProbeTimeout(),
+                            properties.getLivenessProbeDelay(), properties.getLivenessProbePeriod(), "liveness",
+                            request.getDeploymentProperties()).create());
+        }
+
+        if (properties.getReadinessProbePort() == null && properties.getLivenessProbePort() == null && port != null) {
```

Comment:
This third block can likely be removed. The first two conditionals could check the custom probe port first, if not the default port. Something like:

```
Integer readinessProbePort = properties.getReadinessProbePort() != null ? properties.getReadinessProbePort() : port;

if (readinessProbePort != null) { 
...
```

Reply:
Hi @chrisjs. Could you elaborate on the 2nd point? Are you advising looking for the property "spring.cloud.deployer.kubernetes.livenessProbePort" in the AppDeploymentRequest parameter before looking for it in the KubernetesDeployerProperties? Meaning that the ports could be sourced from:
1. default ports
2. AppDeploymentRequest (eg via command line)
3. via the KubernetesDeployerProperties

Reply:
Hi @kmoroe22 yes that sounds correct. Then the probe ports can be defined on the CLI and per app rather than just in the server yaml which would populate `KubernetesDeployerProperties` at startup.